client/src/pages/contentPages/Groups.js

```diff
@@ -1,23 +1,278 @@
+      if (droppedFiles.length) {
+          setUploading(true);
+  
+          const form = new FormData();
+          form.append('image', droppedFiles[0]);
+          const res = await axios.post('/upload/single', form)
+            .catch((err) => console.log(err));
+          if (res && res.data) {
+            setGroupPicture(res.data.imageUrl)
+            setUploading(false)
+          }
+      }
+    }, []);
+  
+    const {
+      getRootProps,
+      isDragActive,
+      isDragAccept,
+      isDragReject,
+    } = useDropzone({onDrop, maxFiles: 1, accept: 'image/*'});
+  
+    const style = useMemo(() => ({
+      ...baseStyle,
+      ...(isDragActive ? activeStyle : {}),
+      ...(isDragAccept ? acceptStyle : {}),
+      ...(isDragReject ? rejectStyle : {})
+    }), [
+      isDragActive,
+      isDragReject,
+      isDragAccept
+    ]);
+
+    const handleOpen = () => {
+      setOpen(true);
+      setGroupPicture('');
+    };
+  
+    const handleClose = () => {
+      setOpen(false);
+      setGroupPicture('');
+      setGroupName('');
+      setGroupErrors('');
+    };
+    
+    const handleSubmit = async (e) => {
+      e.preventDefault();
+      
+      if(isFormValid()) {
+  
+        const data = { groupName: groupName, imageUrl: groupPicture , courseId: courseId};
+        
+        const response = await axios.post('/user/groups', data);
+
+        setGroupName('');
+        setCourseId('');
+        setGroupPicture('');
+  
+        if(response.data) {
+
+          groups.push(response.data.data)
+          actions.fetchUserGroups(groups)(dispatch);
+
+          enqueueSnackbar('Group Created Successfully.', {
+            variant: 'success',
+            autoHideDuration: '5000',
+          });
+          setOpen(false)
+        } else {
+          enqueueSnackbar('Error Creating Group.', {
+            variant: 'Error',
+            autoHideDuration: '5000',
+          });
+        }
+        
+      } else {
+        enqueueSnackbar('Error Creating Group.', {
+            variant: 'Error',
+            autoHideDuration: '5000',
+        });
+      }
+  
+    };
+    
+    const handleOnChange = (e) => {
+      checkGroupExists(e.target.value.toLowerCase())
+      
+    }
+
+    const isFormValid = () => {
+      return (groupNameValid && courseId) ? true : false
+    }
+  
+  const checkGroupExists = (name) => {
+    try {
+      
+      const allGroupNames = [...groups, ...courseGroups].map(group => {
+        return group.name.toLowerCase();
+      })
+
+      setGroupName(name)
+  
+      if(allGroupNames.includes(name) !== true) {
+        setGroupErrors('');
+        setGroupNameValid(true);
+        return false;
+      } else {
+        setGroupErrors("A group with this name already exists!");
+        setGroupNameValid(false);
+        return true;
+      }
+  
+    } catch (err) {
+        console.log(err);
+    }
+  };
+
+  const body = (
+    <div style={modalStyle} className={classes.paper}>
```

Comment:
I see this is where modalStyle is being used, but you could just use multiple classes instead